portfolio/src/main/webapp/js/script.js

```diff
@@ -76,8 +76,19 @@ async function getNumComments(numComments) {
   const messages = await response.json();
   const messageContainer = document.getElementById('message-container');
   messageContainer.innerHTML = '';
-  for (const message of messages) { 
-    messageContainer.appendChild(createListElement(message));
+  document.getElementById('login').setAttribute('href', messages.url);
+  if (JSON.parse(messages.loggedin)) { 
+    document.getElementById('login').innerHTML = 'Logout';
+    for (const message of JSON.parse(messages.comments)) { 
+      if (messages.user == message.useremail) {
+        messageContainer.appendChild(createListElement(message, true));
+      } else {
+        messageContainer.appendChild(createListElement(message, false));
+      }
+    }
+  } else { 
+    document.getElementById('comments-body').innerHTML = `To see and post comments, <a href=${messages.url}>login!</a>`;
```

Comment:
This is quite dangerous and a potential vector for XSS. Please construct the `<a>` element safely with innerText and setting the `href` attribute.